src/System.Net.Security/tests/UnitTests/SslApplicationProtocolTests.cs

```diff
@@ -0,0 +1,96 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Collections.Generic;
+using System.Text;
+using Xunit;
+
+namespace System.Net.Security.Tests
+{
+    public class SslApplicationProtocolTests
+    {
+        [Fact]
+        public void Constants_Values_AreCorrect()
+        {
+            Assert.Equal(new SslApplicationProtocol(new byte[] { 0x68, 0x32 }), SslApplicationProtocol.Http2);
+            Assert.Equal(new SslApplicationProtocol(new byte[] { 0x68, 0x74, 0x74, 0x70, 0x2f, 0x31, 0x2e, 0x31 }), SslApplicationProtocol.Http11);
+        }
+
+        [Fact]
+        public void Constructor_Overloads_Succeeds()
+        {
+            const string hello = "hello";
+            byte[] expected = Encoding.UTF8.GetBytes(hello);
+            SslApplicationProtocol byteProtocol = new SslApplicationProtocol(expected);
+            SslApplicationProtocol stringProtocol = new SslApplicationProtocol(hello);
+            Assert.Equal(byteProtocol, stringProtocol);
+
+            SslApplicationProtocol defaultProtocol = default;
+            Assert.True(defaultProtocol.Protocol.IsEmpty);
+
+            Assert.Throws<ArgumentNullException>(() => { new SslApplicationProtocol((byte[])null); });
+            Assert.Throws<ArgumentNullException>(() => { new SslApplicationProtocol((string)null); });
+            Assert.Throws<ArgumentException>(() => { new SslApplicationProtocol(new byte[] { }); });
+            Assert.Throws<ArgumentException>(() => { new SslApplicationProtocol(string.Empty); });
+            Assert.Throws<EncoderFallbackException>(() => { new SslApplicationProtocol("\uDC00"); });
+        }
+
+        [Theory]
+        [MemberData(nameof(Protocol_Equality_TestData))]
+        public void Equality_Tests_Succeeds(SslApplicationProtocol left, SslApplicationProtocol right)
+        {
+            Assert.Equal(left, right);
+            Assert.True(left == right);
+            Assert.False(left != right);
+
+            if (!left.Protocol.IsEmpty && !right.Protocol.IsEmpty)
```

Comment:
This `if` doesn't belong.  if `==` is true then `GetHashCode()` should match.  And `GetHashCode()` should never throw.